fs/fs.go

```diff
@@ -410,38 +421,107 @@ func (fs *filesystem) fetchSociIndex(ctx context.Context, imageRef, indexDigest,
 		return nil, err
 	}
 
-	var indexDesc ocispec.Descriptor
-	if indexDigest == "" {
-		log.G(ctx).Info("index digest not provided, making a Referrers API call to fetch list of indices")
-		imgDigest, err := digest.Parse(imageManifestDigest)
-		if err != nil {
-			return nil, fmt.Errorf("unable to parse image digest: %w", err)
-		}
+	indexDesc, err := fs.findSociIndexDesc(ctx, imageManifestDigest, indexDigest, remoteStore)
```

Comment:
Naming suggestion: Since with V2, we essentially need to deal with an image index and soci index as well as the image manifests, i think we should follow a naming convention to be clear what we are dealing with. For example, it is not clear if `indexDigest` refers to the soci index Digest or the image index digest. 

First time looking at the code, so may be this is just my learning curve. But using variable names that disamiguate the reference might help with readability.

Reply:
https://github.com/awslabs/soci-snapshotter/issues/1540